### DISCUSSION TITLE
How to control vim mode outside of normal editor contexts

### DISCUSSION THREAD

**Discussion by diminutivesloop** [VSCodeVim/Vim#9101] [Q&A]
I use the [Codeium](https://marketplace.visualstudio.com/items?itemName=Codeium.codeium) and [GitHub PR](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github) extensions. Whenever I edit or add a PR file comment or invoke the Codeium command mode I get vim mode rather than VSCode's default text input mode. I seem to remember there being a setting to control this but I can't find it documented anywhere. Ideally I would like the option to disable this feature in particular contexts or at least set it to default to vim input mode. Is there an option here that I'm missing?

**Update:**
I also noticed that if I enable `vim.smartRelativeLine` I start seeing line numbers in these comment/command inputs which isn't great either.
<img width="1085" alt="Screenshot 2024-06-29 at 4 19 33 PM" src="https://github.com/VSCodeVim/Vim/assets/667716/c0e86d2f-da30-49be-9ef0-fc7234883ebd">

**Comment by 17hz** [comment]
I'm also facing a similar issue and would be grateful for any insights or solutions. It seems that GitHub Copilot doesn't have this problem.
